pkg/controller/remotemachineset/awsactuator.go

```diff
@@ -322,10 +322,10 @@ func (a *AWSActuator) getPrivateSubnetsByAvailabilityZone(pool *hivev1.MachinePo
 
 	results, err := a.awsClient.DescribeSubnets(&ec2.DescribeSubnetsInput{SubnetIds: idPointers})
 	if err != nil || len(results.Subnets) == 0 {
-		if strings.Contains(err.Error(), "InvalidSubnetID.NotFound") {
-			var conditionMessage string
+		if strings.Contains(err.Error(), "InvalidSubnet") {
```

Comment:
I hit `InvalidSubnets` for `subnets specfied in the machinepool do not exist` test - this seemed like a good compromise.
`err.Error()` is `InvalidSubnets`